Automattic/sensei
# Add translations for lessons and quizzes that were created while saving the course structure

Resolves #6978 

When we create a lesson or a quiz while saving the course structure, the translation is not created in WPML. Unfortunately, I didn't find why it works that way specifically for this case: if you create a course or a lesson in UI, everything work fine. I suppose it might happen with loading meta boxes or saving data from them.

## Proposed Changes
* Add translations for lessons and quizzes that were created while saving the course structure

## Testing Instructions

1. Install and activate WPML.
2. Go to Sensei LMS > Courses, create a new course.
3. Start with a blank (or default?) course structure (when 3 default lessons pre-filled). That's for simplicity, and you can play at this point.
4. Save draft or publish the course.
5. It is expected that the status of lessons changed from Unsaved to Draft.
6. You can also check if a translation entry was created in database: `SELECT * FROM wp_icl_translations WHERE element_id=7686; -- 7686 is Lesson ID`

## New/Updated Hooks

* `sensei_course_structure_lesson_created` - Fires after a lesson is created while saving the course structure.
* `sensei_course_structure_quiz_created` - Fires after a quiz is created while saving the course structure.


## Pre-Merge Checklist
